ocamltest/run_unix.c

```diff
@@ -147,6 +147,8 @@ static void update_environment(array local_env)
       memcpy(value, pos_eq + 1, value_length);
       value[value_length] = '\0';
       setenv(name, value, 1); /* 1 means overwrite */
+      free(name);
+      free(value);
```

Comment:
It was not obvious to me, but some [Linux manpages](https://man7.org/linux/man-pages/man3/setenv.3.html) state

> This function makes copies of the strings pointed to by name and value (by contrast with putenv(3)).

and the [OpenGroup](https://pubs.opengroup.org/onlinepubs/009695399/functions/setenv.html) manpage also states that the strings are copied, so indeed they can and should be freed afterwards.

Reply:
I’ve updated the commit message to include your post and references to the manpages.